### ISSUE TITLE
Add ARN attribute for aws_instance

### ISSUE THREAD

**Issue by tdmalone** [terraform-providers/terraform-provider-aws#5425]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I'd love to be able to directly access the ARN of an EC2 instance, for example for interpolation into an IAM policy.

### New or Affected Resource(s)

* [`aws_instance`](https://www.terraform.io/docs/providers/aws/r/instance.html)
* [`aws_instance`](https://www.terraform.io/docs/providers/aws/d/instance.html) (data resource)

Of note, the ARN is already exported for the [`db_instance`](https://www.terraform.io/docs/providers/aws/r/db_instance.html) resource and [data resource](https://www.terraform.io/docs/providers/aws/d/db_instance.html).

### Potential Terraform Configuration

```hcl
resource "aws_instance" "main" {
  // ... etc. etc.
}

data "template_file" "iam_policy_template" {
  template = "$(file("my-policy.json")}"

  vars {
    instance_arn = "${aws_instance.main.arn}"
  }
}
```

**Comment by blckct** [comment]
That's probably not the only resource with missing ARN, probably would be a good idea to go over them and check.

**Comment by bflad** [comment]
The new `arn` attribute for the resource and data source has been merged and will release with version 1.31.0 of the AWS provider later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.31.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1310-august-09-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
